src/main/java/com/manywho/services/saml/services/AuthenticationService.java

```diff
@@ -26,6 +26,19 @@ public AuthenticatedWhoResult createAuthenticatedWhoResultWithError(String error
     }
 
     public AuthenticatedWhoResult createAuthenticatedWhoResult(SamlResponseHandler response) throws Exception {
+
+        if (response.isValid() == false) {
+            return createAuthenticatedWhoResultWithError(response.getError());
+        }
+
+        String jwtToken = jwtService.sign(response.getNameIdentifier(), response.getResponse().getNotBefore(), response.getResponse().getNotAfter());
```

Comment:
Could be worth placing this within the following try-catch - so we send a response immediately in case of an error, instead of just throwing a generic error

Reply:
Okey. The only thing that I expect to fail here are tokens with expired date, but lets put everything in the try-catch just in case.